### DISCUSSION TITLE
Change the vite mode in component testing

### DISCUSSION THREAD

**Discussion by RobbeCl** [cypress-io/cypress#28371] [Component Testing]
How do I pass Vite flags (like mode: https://vitejs.dev/config/shared-options.html#mode) through the cypress run command?

When I change my configuration file from
```
  component: {
    devServer: {
      framework: 'vue',
      bundler: 'vite',
    },
  }

```
to
```
  component: {
    devServer: {
      framework: 'vue',
      bundler: 'vite',
      viteConfig: {
        shared: {
          mode: 'msw',
        },
      },
    },
  }
```

My imports fail to resolve ([plugin:vite:import-analysis] Failed to resolve import "@/components/Typography" from "cypress/support/component.ts". Does the file exist?)

I think it suddenly stopped taking my default vite.config.ts file into account.

**Comment by M165437** [comment]
See this change: https://www.cypress.io/blog/2022/11/04/upcoming-changes-to-component-testing#vite-dev-server

> In Cypress 11, if you provide an inline viteConfig in your configuration file, your vite.config will not be merged automatically. The inline viteConfig will be considered the source of truth. If you want to merge the two configurations, you can simply import and merge them manually.
